server/src/assets/game/tableGenerator/converter.py

```diff
@@ -1,127 +1,162 @@
 from csv import reader
-import json
-import yaml
-
-interlockingTableFile = "../../../../../configurations/swtbahn-full/interlocking_table.yml"
-configuratonBahnFile = "../../../../../configurations/swtbahn-full/extras_config.yml"
-blacklistFile = "./blacklist.txt"
-groupingFile = "./flags-swtbahn-full.csv"
-interlockingTable = json.loads(json.dumps(yaml.safe_load(open(interlockingTableFile))))
-configuratonBahn = json.loads(json.dumps(yaml.safe_load(open(configuratonBahnFile))))
-
-blacklist = []
-
-for line in open(blacklistFile):
-    blacklist.append(int(line.strip()))
-
-resultData = {}
-
-blocktypes = ["blocks", "platforms"]
-
-for blockType in blocktypes:
-    for block in configuratonBahn[blockType]:
-        try:
-            signals = block["signals"]
-        except KeyError:
-#            print("Block {} has no signals".format(block["id"]))
-            signals = []
-        routes = []
-        for signal in signals:
-            destinations = []
-            for route in interlockingTable["interlocking-table"]:
-                if route["source"] == signal:
-                    if route["destination"] not in destinations:
-                        destinations.append(route["destination"])
-            for destination in destinations:
-                routeID = -1
+import json, yaml, os
+
```

Comment:
a few more comments across the script as a whole would be nice.   
At some points, there are a LOT of nested for-loops. Think about - perhaps - breaking this down into more separate functions.